clang/test/Driver/sycl-offload.cpp

```diff
@@ -47,3 +47,60 @@
+/// forced from command line.
+// RUN:  echo "void foo();" > %t_dummy.cpp
+// RUN:  %clang -fsycl -c %t_dummy.cpp -o %t_dummy.o
+// RUN:  llvm-ar cr %t_dummy.a %t_dummy.o
+// RUN:  touch %t_empty.o
+// RUN:  %clangxx -### -fsycl -fsycl-targets=spir64_x86_64 %t_dummy.o %s 2>&1 \
+// RUN:    | FileCheck -check-prefix IMPLIED_DEVICE_OBJ %s
+// RUN:  %clangxx -### -fsycl -fsycl-targets=spir64_fpga %t_dummy.o %s 2>&1 \
+// RUN:    | FileCheck -check-prefix IMPLIED_DEVICE_OBJ %s
+// RUN:  %clangxx -### -fsycl -fsycl-targets=spir64_gen %t_dummy.o %s 2>&1 \
+// RUN:    | FileCheck -check-prefix IMPLIED_DEVICE_OBJ %s
+// RUN:  %clangxx -### -fsycl -fintelfpga %t_dummy.o %s 2>&1 \
+// RUN:    | FileCheck -check-prefix IMPLIED_DEVICE_OBJ %s
+// RUN:  %clangxx -### -fsycl -fsycl-add-spirv -fsycl-targets=spir64_x86_64 %t_empty.o %s 2>&1 \
+// RUN:    | FileCheck -check-prefix IMPLIED_DEVICE_OBJ %s
+// RUN:  %clangxx -### -fsycl -fsycl-add-spirv -fsycl-targets=spir64_fpga %t_empty.o %s 2>&1 \
+// RUN:    | FileCheck -check-prefix IMPLIED_DEVICE_OBJ %s
+// RUN:  %clangxx -### -fsycl -fsycl-add-spirv -fsycl-targets=spir64_gen %t_empty.o %s 2>&1 \
+// RUN:    | FileCheck -check-prefix IMPLIED_DEVICE_OBJ %s
+// RUN:  %clangxx -### -fsycl -fsycl-add-spirv -fintelfpga %t_empty.o %s 2>&1 \
+// RUN:    | FileCheck -check-prefix IMPLIED_DEVICE_OBJ %s
+// IMPLIED_DEVICE_OBJ: clang-offload-bundler{{.*}} "-type=o"{{.*}} "-targets=sycl-spir64-unknown-unknown-sycldevice,sycl-spir64_{{.*}}-unknown-unknown-sycldevice"{{.*}} "-unbundle"
+
+// RUN:  %clangxx -### -fsycl -fsycl-targets=spir64_x86_64 %t_dummy.a %s 2>&1 \
+// RUN:    | FileCheck -check-prefix IMPLIED_DEVICE_LIB %s
+// RUN:  %clangxx -### -fsycl -fsycl-targets=spir64_fpga %t_dummy.a %s 2>&1 \
+// RUN:    | FileCheck -check-prefix IMPLIED_DEVICE_LIB %s
+// RUN:  %clangxx -### -fsycl -fsycl-targets=spir64_gen %t_dummy.a %s 2>&1 \
+// RUN:    | FileCheck -check-prefix IMPLIED_DEVICE_LIB %s
+// RUN:  %clangxx -### -fsycl -fintelfpga %t_dummy.a %s 2>&1 \
+// RUN:    | FileCheck -check-prefix IMPLIED_DEVICE_LIB %s
+// IMPLIED_DEVICE_LIB: clang-offload-bundler{{.*}} "-type=a"{{.*}} "-targets=sycl-spir64-unknown-unknown-sycldevice,sycl-spir64_{{.*}}-unknown-unknown-sycldevice"{{.*}} "-unbundle"
+
+/// Check that the default device triple is not used with -fno-sycl-link-spirv
+// RUN:  %clangxx -### -fsycl -fno-sycl-link-spirv -fsycl-targets=spir64_x86_64 %t_dummy.o %s 2>&1 \
+// RUN:    | FileCheck -check-prefix NO_IMPLIED_DEVICE_OPT %s
+// RUN:  %clangxx -### -fsycl -fno-sycl-link-spirv -fsycl-targets=spir64_fpga %t_dummy.o %s 2>&1 \
+// RUN:    | FileCheck -check-prefix NO_IMPLIED_DEVICE_OPT %s
+// RUN:  %clangxx -### -fsycl -fno-sycl-link-spirv -fsycl-targets=spir64_gen %t_dummy.o %s 2>&1 \
+// RUN:    | FileCheck -check-prefix NO_IMPLIED_DEVICE_OPT %s
+// RUN:  %clangxx -### -fsycl -fno-sycl-link-spirv -fintelfpga %t_dummy.o %s 2>&1 \
+// RUN:    | FileCheck -check-prefix NO_IMPLIED_DEVICE_OPT %s
```

Comment:
It may be worthwhile checking that the "intended" triple is still being added for `-fno-sycl-link-spirv` cases.